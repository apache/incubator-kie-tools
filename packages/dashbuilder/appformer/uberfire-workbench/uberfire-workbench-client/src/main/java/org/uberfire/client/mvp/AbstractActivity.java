/*
 * Copyright 2015 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.uberfire.client.mvp;

import org.uberfire.mvp.PlaceRequest;

import static org.kie.soup.commons.validation.PortablePreconditions.checkNotNull;

/**
 * Implementation of behaviour common to all activity types.
 * <p>
 * AbstractActivity is not typically subclassed directly, even by generated code. See the more purpose-based subclasses.
 */
public abstract class AbstractActivity implements Activity {

    protected PlaceManager placeManager;

    protected PlaceRequest place;

    protected boolean open = false;

    public AbstractActivity(final PlaceManager placeManager) {
        this.placeManager = placeManager;
    }

    /**
     * Tracks start/shutdown lifecycle. Subclasses should always call <tt>super.onStartup()</tt> in methods that
     * override this one.
     */
    @Override
    public void onStartup(PlaceRequest place) {
        this.place = checkNotNull("place",
                                  place);
    }

    /**
     * Tracks open/closed lifecycle. Subclasses should always call <tt>super.onOpen()</tt> in methods that override this
     * one.
     */
    @Override
    public void onOpen() {
        if (this.place == null) {
            throw new IllegalStateException("Activity " + this + " has not been started");
        }
        if (open) {
            throw new IllegalStateException("Activity " + this + " already open");
        }
        open = true;
        placeManager.executeOnOpenCallbacks(this.place);
    }

    /**
     * Tracks open/closed lifecycle. Subclasses should always call <tt>super.onClose()</tt> in methods that override
     * this one.
     */
    @Override
    public void onClose() {
        if (this.place == null) {
            throw new IllegalStateException("Activity " + this + " has not been started");
        }
        if (!open) {
            throw new IllegalStateException("Activity " + this + " not open");
        }
        open = false;
        placeManager.executeOnCloseCallbacks(this.place);
    }

    /**
     * Tracks start/shutdown lifecycle. Subclasses should always call <tt>super.onShutdown()</tt> in methods that
     * override this one.
     */
    @Override
    public void onShutdown() {
        if (this.place == null) {
            throw new IllegalStateException("Activity " + this + " has not been started");
        }
        if (open) {
            throw new IllegalStateException("Activity " + this + " is open");
        }
        this.place = null;
    }

    @Override
    public PlaceRequest getPlace() {
        return place;
    }

    @Override
    public String toString() {
        return getClass().getName() + (place == null ? " (not started)" : " for " + place);
    }
}
