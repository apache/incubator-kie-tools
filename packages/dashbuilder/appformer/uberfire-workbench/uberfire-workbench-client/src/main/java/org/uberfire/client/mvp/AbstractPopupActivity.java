/*
 * Copyright 2015 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.uberfire.client.mvp;

import com.google.gwt.event.logical.shared.CloseEvent;
import com.google.gwt.event.logical.shared.CloseHandler;
import com.google.gwt.user.client.ui.IsWidget;
import org.uberfire.client.annotations.WorkbenchPopup;
import org.uberfire.client.annotations.WorkbenchPopup.WorkbenchPopupSize;
import org.uberfire.client.workbench.widgets.popup.PopupView;
import org.uberfire.security.ResourceType;
import org.uberfire.workbench.model.ActivityResourceType;

/**
 * Implementation of behaviour common to all popup activities. Concrete implementations are typically not written by
 * hand; rather, they are generated from classes annotated with {@link WorkbenchPopup}.
 */
public abstract class AbstractPopupActivity extends AbstractActivity implements PopupActivity {

    private final PopupView popup;

    /**
     * Interlock to prevent a call back into PlaceManager.closePlace() when this activity is already in the process of
     * closing.
     */
    private boolean placeManagerIsClosingUs = false;

    /**
     * Interlock to prevent a call back into PopupView.hide() when the view is already in the process of hiding.
     */
    private boolean popupAlreadyHiding = false;

    /**
     * MVP constructor that allows caller to provide the PopupView instance.
     */
    protected AbstractPopupActivity(final PlaceManager placeManager,
                                    final PopupView popupView) {
        super(placeManager);
        popup = popupView;
    }

    @Override
    public ResourceType getResourceType() {
        return ActivityResourceType.POPUP;
    }

    @Override
    public abstract String getTitle();

    @Override
    public WorkbenchPopupSize getSize() {
        return WorkbenchPopupSize.MEDIUM;
    }

    @Override
    public IsWidget getTitleDecoration() {
        return null;
    }

    @Override
    public abstract IsWidget getWidget();

    @Override
    public void onOpen() {
        super.onOpen();

        popup.addCloseHandler(new CloseHandler<PopupView>() {
            @Override
            public void onClose(CloseEvent<PopupView> event) {
                if (!placeManagerIsClosingUs) {
                    try {
                        popupAlreadyHiding = true;
                        placeManager.closePlace(place);
                    } finally {
                        popupAlreadyHiding = false;
                    }
                }
            }
        });

        final IsWidget widget = getWidget();

        popup.setContent(widget);
        popup.setSize(getSize());
        popup.setTitle(getTitle());
        popup.show();
    }

    @Override
    public void onClose() {
        super.onClose();
        if (!popupAlreadyHiding) {
            try {
                placeManagerIsClosingUs = true;
                popup.hide();
            } finally {
                placeManagerIsClosingUs = false;
            }
        }
    }

    @Override
    public boolean onMayClose() {
        return true;
    }
}
