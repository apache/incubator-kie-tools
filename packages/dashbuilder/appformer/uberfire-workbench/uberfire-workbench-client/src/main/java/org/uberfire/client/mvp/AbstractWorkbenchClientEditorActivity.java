/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.uberfire.client.mvp;

import org.uberfire.client.annotations.WorkbenchClientEditor;

/**
 * Implementation of behaviour common to all workbench editor activities. Concrete implementations are typically not
 * written by hand; rather, they are generated from classes annotated with {@link WorkbenchClientEditor}.
 */
public abstract class AbstractWorkbenchClientEditorActivity extends AbstractWorkbenchActivity implements WorkbenchClientEditorActivity {

    public AbstractWorkbenchClientEditorActivity(final PlaceManager placeManager) {
        super(placeManager);
    }
}