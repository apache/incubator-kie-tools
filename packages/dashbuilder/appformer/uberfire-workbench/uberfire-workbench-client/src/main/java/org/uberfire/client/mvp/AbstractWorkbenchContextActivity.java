/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 * 
 *  http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License. 
 */

package org.uberfire.client.mvp;

import com.google.gwt.user.client.ui.IsWidget;
import org.uberfire.client.annotations.WorkbenchContext;
import org.uberfire.workbench.model.PanelDefinition;

/**
 * Implementation of behaviour common to all context activities. Concrete implementations are typically not written by
 * hand; rather, they are generated from classes annotated with {@link WorkbenchContext}.
 */
public abstract class AbstractWorkbenchContextActivity extends AbstractActivity implements ContextActivity {

    protected PanelDefinition panel;

    public AbstractWorkbenchContextActivity(final PlaceManager placeManager) {
        super(placeManager);
    }

    @Override
    public void onAttach(final PanelDefinition panel) {
        this.panel = panel;
    }

    @Override
    public IsWidget getTitleDecoration() {
        return null;
    }
}
