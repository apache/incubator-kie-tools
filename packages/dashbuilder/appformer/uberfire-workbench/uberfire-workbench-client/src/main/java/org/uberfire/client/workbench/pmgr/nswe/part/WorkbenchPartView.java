/*
 * Copyright 2015 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.uberfire.client.workbench.pmgr.nswe.part;

import javax.enterprise.context.Dependent;

import com.google.gwt.user.client.ui.IsWidget;
import com.google.gwt.user.client.ui.ScrollPanel;
import com.google.gwt.user.client.ui.SimpleLayoutPanel;
import org.uberfire.client.util.Layouts;
import org.uberfire.client.workbench.part.WorkbenchPartPresenter;

/**
 * A Workbench panel part.
 */
@Dependent
public class WorkbenchPartView
        extends SimpleLayoutPanel
        implements WorkbenchPartPresenter.View {

    private final ScrollPanel sp = new ScrollPanel();
    private WorkbenchPartPresenter presenter;

    public WorkbenchPartView() {
        setWidget(sp);

        // ScrollPanel creates an additional internal div that we need to style
        sp.getElement().getFirstChildElement().setClassName("uf-scroll-panel");
    }

    @Override
    public void init(WorkbenchPartPresenter presenter) {
        this.presenter = presenter;
        Layouts.setToFillParent(this);
    }

    @Override
    public WorkbenchPartPresenter getPresenter() {
        return this.presenter;
    }

    @Override
    public IsWidget getWrappedWidget() {
        return sp.getWidget();
    }

    @Override
    public void setWrappedWidget(final IsWidget widget) {
        sp.setWidget(widget);
    }
}
