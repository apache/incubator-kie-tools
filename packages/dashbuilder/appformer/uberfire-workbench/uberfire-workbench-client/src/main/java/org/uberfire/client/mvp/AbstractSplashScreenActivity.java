/*
 * Copyright 2015 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.uberfire.client.mvp;

import javax.annotation.PostConstruct;
import javax.inject.Inject;

import com.google.gwt.event.logical.shared.CloseEvent;
import com.google.gwt.event.logical.shared.CloseHandler;
import com.google.gwt.user.client.ui.IsWidget;
import org.uberfire.client.annotations.WorkbenchSplashScreen;
import org.uberfire.client.workbench.WorkbenchServicesProxy;
import org.uberfire.client.workbench.widgets.splash.SplashView;
import org.uberfire.mvp.ParameterizedCommand;
import org.uberfire.mvp.PlaceRequest;
import org.uberfire.workbench.model.SplashScreenFilter;

/**
 * Implementation of behaviour common to all splash screen activities. Concrete implementations are typically not written by
 * hand; rather, they are generated from classes annotated with {@link WorkbenchSplashScreen}.
 */
public abstract class AbstractSplashScreenActivity extends AbstractActivity implements SplashScreenActivity {

    private final SplashView splash;
    @Inject
    private WorkbenchServicesProxy wbServices;
    private Boolean showAgain;
    private SplashScreenFilter splashFilter;

    @Inject
    public AbstractSplashScreenActivity(final PlaceManager placeManager,
                                        final SplashView splash) {
        super(placeManager);
        this.splash = splash;
    }

    @PostConstruct
    private void initialize() {
        this.splashFilter = getFilter();
        splash.addCloseHandler(new CloseHandler<SplashView>() {
            @Override
            public void onClose(final CloseEvent<SplashView> event) {
                AbstractSplashScreenActivity.this.onClose();
            }
        });
    }

    @Override
    public void onOpen() {
        super.onOpen();
    }

    @Override
    public void onStartup(PlaceRequest place) {
        super.onStartup(place);
        wbServices.loadSplashScreenFilter(getFilter().getName(),
                                          new ParameterizedCommand<SplashScreenFilter>() {
                                              @Override
                                              public void execute(final SplashScreenFilter response) {
                                                  if (response != null) {
                                                      splashFilter = response;
                                                  }
                                                  if (splashFilter.displayNextTime()) {
                                                      forceShow();
                                                  }
                                              }
                                          });
    }

    @Override
    public abstract String getTitle();

    @Override
    public abstract IsWidget getWidget();

    @Override
    public Integer getBodyHeight() {
        return null;
    }

    @Override
    public abstract SplashScreenFilter getFilter();

    @Override
    public IsWidget getTitleDecoration() {
        return null;
    }

    @Override
    public void closeIfOpen() {
        if (splash.isAttached()) {
            splash.hide();
            onClose();
        }
    }

    @Override
    public void forceShow() {
        final IsWidget widget = getWidget();

        splash.setContent(widget,
                          getBodyHeight());
        splash.setTitle(getTitle());
        splash.show();
    }

    @Override
    public void onClose() {
        saveState();
        super.onClose();
    }

    @Override
    public Boolean intercept(final PlaceRequest intercepted) {
        if (splashFilter == null) {
            return false;
        }
        for (final String interceptPoint : splashFilter.getInterceptionPoints()) {
            if (intercepted.getIdentifier().equals(interceptPoint)) {
                return true;
            }
        }

        return false;
    }

    private void saveState() {
        showAgain = splash.showAgain();
        if (showAgain != null) {
            splashFilter.setDisplayNextTime(showAgain);
            wbServices.save(splashFilter);
        }
    }
}