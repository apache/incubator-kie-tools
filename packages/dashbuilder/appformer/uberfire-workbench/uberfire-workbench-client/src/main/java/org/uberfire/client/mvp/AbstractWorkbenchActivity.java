/*
 * Copyright 2015 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.uberfire.client.mvp;

import java.util.function.Consumer;

import com.google.gwt.user.client.ui.IsWidget;
import org.uberfire.mvp.PlaceRequest;
import org.uberfire.workbench.model.CompassPosition;
import org.uberfire.workbench.model.Position;
import org.uberfire.workbench.model.menu.Menus;
import org.uberfire.workbench.model.toolbar.ToolBar;

/**
 * Implementation of behaviour common to all workbench editors and screens.
 * <p>
 * AbstractWorkbenchActivity is not typically subclassed directly, even by generated code. See the more purpose-based
 * subclasses.
 */
public abstract class AbstractWorkbenchActivity extends AbstractActivity implements WorkbenchActivity {

    public AbstractWorkbenchActivity(final PlaceManager placeManager) {
        super(placeManager);
    }

    @Override
    public Position getDefaultPosition() {
        return CompassPosition.ROOT;
    }

    /**
     * This default implementation returns null, meaning this activity can be displayed in any perspective.
     */
    @Override
    public PlaceRequest getOwningPlace() {
        return null;
    }

    @Override
    public boolean onMayClose() {
        return true;
    }

    @Override
    public abstract String getTitle();

    @Override
    public IsWidget getTitleDecoration() {
        return null;
    }

    @Override
    public abstract IsWidget getWidget();

    @Override
    public void onLostFocus() {
        //Do nothing.
    }

    @Override
    public void onFocus() {
    }

    @Override
    public void getMenus(final Consumer<Menus> menusConsumer) {
        menusConsumer.accept(null);
    }

    @Override
    public ToolBar getToolBar() {
        return null;
    }

    @Override
    public String contextId() {
        return null;
    }
}
