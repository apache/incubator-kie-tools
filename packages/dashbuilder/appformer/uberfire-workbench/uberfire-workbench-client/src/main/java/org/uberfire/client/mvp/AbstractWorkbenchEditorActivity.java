/*
 * Copyright 2015 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.uberfire.client.mvp;

import org.uberfire.backend.vfs.ObservablePath;
import org.uberfire.backend.vfs.Path;
import org.uberfire.client.annotations.WorkbenchEditor;
import org.uberfire.client.annotations.WorkbenchEditor.LockingStrategy;
import org.uberfire.mvp.PlaceRequest;
import org.uberfire.mvp.impl.PathPlaceRequest;

/**
 * Implementation of behaviour common to all workbench editor activities. Concrete implementations are typically not
 * written by hand; rather, they are generated from classes annotated with {@link WorkbenchEditor}.
 */
public abstract class AbstractWorkbenchEditorActivity extends AbstractWorkbenchActivity implements WorkbenchEditorActivity {

    protected ObservablePath path;

    public AbstractWorkbenchEditorActivity(final PlaceManager placeManager) {
        super(placeManager);
    }

    /**
     * Overrides the default implementation by redirecting calls that are {@link PathPlaceRequest} instances to
     * {@link #onStartup(ObservablePath, PlaceRequest)}. Non-path place requests are handed up to the super impl.
     */
    @Override
    public final void onStartup(final PlaceRequest place) {
        final Path path = place.getPath();
        if (path != null) {
            if (path instanceof ObservablePath) {
                onStartup((ObservablePath) path,
                          place);
            } else if (this.isDynamic()) {
                onStartup(path,
                          place);
            }
        } else {
            super.onStartup(place);
        }
    }

    void onStartup(final Path path,
                   final PlaceRequest place) {

        onStartup(new PathPlaceRequest(path).getPath(),
                  place);
    }

    @Override
    public void onStartup(final ObservablePath path,
                          final PlaceRequest place) {
        super.onStartup(place);
        this.path = path;

    }


    @Override
    public void onOpen() {
        super.onOpen();
    }

    @Override
    public void onSave() {
        //Do nothing.
    }

    @Override
    public boolean isDirty() {
        return false;
    }

    @Override
    public void onClose() {
        super.onClose();
    }

    @Override
    public void onFocus() {
        super.onFocus();
    }


    /**
     * Returns the locking strategy for this editor activity, defaulting to
     * pessimistic. This method is overridden for generated activities returning
     * the strategy configured at {@link WorkbenchEditor}.
     */
    protected LockingStrategy getLockingStrategy() {
        return LockingStrategy.FRAMEWORK_PESSIMISTIC;
    }
}