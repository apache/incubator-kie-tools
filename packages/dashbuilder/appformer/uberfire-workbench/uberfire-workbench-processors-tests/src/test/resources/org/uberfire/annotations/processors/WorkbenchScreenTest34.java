package org.uberfire.annotations.processors;

import com.google.gwt.user.client.ui.IsWidget;
import com.google.gwt.user.client.ui.SimplePanel;
import org.uberfire.client.annotations.WorkbenchPartTitle;
import org.uberfire.client.annotations.WorkbenchPartView;
import org.uberfire.client.annotations.WorkbenchScreen;
import org.uberfire.client.mvp.QualifierAnnotation;
import org.uberfire.client.mvp.RegularAnnotation;

@QualifierAnnotation( classField = String.class, stringField = "someText", booleanField = true, intField = 13 )
@RegularAnnotation( classField = String.class, stringField = "someText", booleanField = true, intField = 13 )
@WorkbenchScreen(identifier = "test34")
public class WorkbenchScreenTest34 {

    @WorkbenchPartView
    public IsWidget getView() {
        return new SimplePanel();
    }
    
    @WorkbenchPartTitle
    public String getTitle() {
        return "title";
    }

}
