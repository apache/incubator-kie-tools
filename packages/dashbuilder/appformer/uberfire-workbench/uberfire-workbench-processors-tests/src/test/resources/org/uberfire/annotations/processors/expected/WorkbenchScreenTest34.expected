/*
 * Copyright 2012 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.uberfire.annotations.processors;

import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.function.Consumer;
import javax.annotation.Generated;
import javax.enterprise.context.Dependent;
import javax.inject.Inject;
import com.google.gwt.user.client.ui.InlineLabel;

import javax.inject.Named;
import org.uberfire.client.mvp.AbstractWorkbenchScreenActivity;
import org.uberfire.client.mvp.PlaceManager;

import org.uberfire.mvp.PlaceRequest;

import com.google.gwt.user.client.ui.IsWidget;

@Dependent
@Generated("org.uberfire.annotations.processors.WorkbenchScreenProcessor")
@Named("test34")
@org.uberfire.client.mvp.QualifierAnnotation(classField=java.lang.String.class, stringField="someText", booleanField=true, intField=13)
/*
 * WARNING! This class is generated. Do not modify.
 */
public class WorkbenchScreenTest34Activity extends AbstractWorkbenchScreenActivity {

    @Inject
    @org.uberfire.client.mvp.QualifierAnnotation(classField=java.lang.String.class, stringField="someText", booleanField=true, intField=13)
    private WorkbenchScreenTest34 realPresenter;

    @Inject
    //Constructor injection for testing
    public WorkbenchScreenTest34Activity(final PlaceManager placeManager) {
        super( placeManager );
    }

    @Override
    public String getTitle() {
        return realPresenter.getTitle();
    }

    @Override
    public IsWidget getWidget() {
        return realPresenter.getView();
    }

    @Override
    public String getIdentifier() {
        return "test34";
    }
}
